### ISSUE TITLE
Обоснование выбора YOLOv8 и описание используемых технологий

### ISSUE THREAD

**Issue by 01-Zhangsaisai** [st091150/Agricultural-robot-modules#54]
Почему YOLOv8 была выбрана для задачи детекции пшеницы и сорняков
YOLOv8 предоставляет оптимальный баланс между качеством детекции и скоростью работы, что делает её подходящей для агротехнических задач в реальном времени. Основные причины выбора:
Высокая точность на малых и средних объектах, что важно для детекции сорняков между злаками.
Небольшой размер модели (YOLOv8n) — подходит для встраиваемых систем робототехники.
Простота обучения и инференса: минимальная подготовка среды, удобный CLI/API.
Встроенная аугментация данных, улучшающая результаты при небольшом объёме датасета.
Активно поддерживаемая экосистема Ultralytics.
Как проводилось обучение модели
Подготовка датасета: структура images/, labels/, разметка в формате YOLO.
Создание конфигурационного файла cwfid.yaml с путями к данным и классами (crop, weed).
Обучение модели командой:
yolo detect train model=yolov8n.pt data=cwfid.yaml epochs=150 imgsz=640
Отслеживание метрик: precision, recall, mAP50, mAP50-95.
Выбор лучшей модели (best.pt) по валидационному mAP.
Использованные библиотеки и инструменты
Основные технологии проекта:
Ultralytics YOLOv8 — обучение и инференс модели.
PyTorch — глубинное обучение и вычисления.
OpenCV — загрузка изображений и подготовка данных.
NumPy — работа с массивами и разметкой.
PyYAML — чтение и запись .yaml файлов.
Matplotlib — визуализация разметки и результатов (при необходимости).
Python 3.10 — основная версия интерпретатора.
Дополнительные утилиты:
Скрипты: convert_to_yolo.py, split_dataset.py.
Структура проекта адаптирована под стандарты Ultralytics.
Ожидаемый результат
Документирован выбор архитектуры YOLOv8.
Полное описание используемых библиотек.
Чёткая фиксация процесса обучения модели для дальнейшего воспроизведения.
